pkg/reconciler/apiserversource/controller.go

```diff
@@ -57,9 +65,17 @@ func NewController(
 		Base:                  reconciler.NewBase(ctx, controllerAgentName, cmw),
 		apiserversourceLister: apiServerSourceInformer.Lister(),
 		deploymentLister:      deploymentInformer.Lister(),
+		eventTypeLister:       eventTypeInformer.Lister(),
 		source:                GetCfgHost(ctx),
 		loggingContext:        ctx,
 	}
+
+	env := &envConfig{}
+	if err := envconfig.Process("", env); err != nil {
+		r.Logger.Panicf("unable to process APIServerSource's required environment variables: %v", err)
```

Comment:
With the old code, I think we only panic'ed once we needed the receive adapter. With this change, I think we will panic during startup. I tend to prefer this way, but I think the other was chosen for a reason. @n3wscott do you remember why?

Reply:
@n3wscott do you remember why?

Reply:
I cannot think of any reasons other than not crashing when the env variable is not provided and the apiserver source is not used.

BTW I think I wrote this code (more than 6 months ago)

@Harwayne @n3wscott